src/components/FileInput/index.tsx

```diff
@@ -0,0 +1,198 @@
+        errorContainerClassName,
+        hint,
+        hintContainerClassName,
+        icons,
+        iconsContainerClassName,
+        inputSectionClassName,
+        inputContainerClassName,
+        label,
+        labelContainerClassName,
+        readOnly,
+        uiMode,
+        inputElementRef,
+        containerRef,
+        inputSectionRef,
+        inputClassName,
+        value,
+        onChange,
+        showStatus = true,
+        name,
+        multiple,
+        accept,
+        ...fileInputProps
+    } = props;
+
+    const uiModeClassName = useUiModeClassName(uiMode, styles.light, styles.dark);
+
+    const [status, setStatus] = useState<string>('No file chosen');
+    const isValidFile = useCallback((fileName: string, mimeType: string, acceptString?: string) => {
+        // if there is no accept string, anything is valid
+        if (!acceptString) {
+            return true;
+        }
+        const extensionMatch = /\.\w+$/.exec(fileName);
+        const mimeMatch = /^.+\//.exec(mimeType);
+
+        const fileTypeList = acceptString.split(/,\s+/);
+        return fileTypeList.some((fileType) => {
+            // check mimeType such as image/png or image/*
+            if (mimeType === fileType || (!!mimeMatch && `${mimeMatch[0]}*` === fileType)) {
+                return true;
+            }
+            return !!extensionMatch && extensionMatch[0].toLowerCase() === fileType.toLowerCase();
+        });
+    }, []);
+
+    const getStatus = useCallback((files: File[]) => {
+        if (files && files.length > 1) {
+            return `${files?.length} files selected`;
+        }
+        if (files && files.length === 0) {
+            return 'No file chosen';
+        }
+        if (files) {
+            return files[0]?.name;
+        }
+        return 'No file chosen';
+    }, []);
+
+    const handleChange = useCallback(
+        (_: string | undefined, __: T, e?: React.FormEvent<HTMLInputElement>) => {
+            if (e) {
+                const { files: newFiles } = (e.target as HTMLInputElement);
+                if (newFiles) {
+                    const fileList = Array.from(newFiles);
+                    const validFiles = fileList.filter((f) => isValidFile(f.name, f.type, accept));
+                    const newStatus = getStatus(validFiles);
+                    setStatus(newStatus);
+                    if (onChange) {
+                        onChange(validFiles);
+                    }
+                }
+            }
+        }, [onChange, isValidFile, accept, getStatus],
+    );
+
+    const handleDrop: React.DragEventHandler<HTMLDivElement> = useCallback((e) => {
+        e.preventDefault();
+        const fileList = Array.from(e.dataTransfer.files);
+        const validFiles = fileList.filter((f) => isValidFile(f.name, f.type, accept));
+        if (!multiple && fileList.length > 1) {
+            setStatus('Multiple file selection not allowed. Please, select a single file.');
+        } else {
```

Comment:
We should add this check to handleChange as well (to be 100% certain)

Reply:
isn't the ```multiple``` prop passed to ```<input type='file'>``` control number of files that can be selected?. If so I think the condition won't arise in handleChange.

Reply:
It should. But, it just depends on each browser. Better be safe than sorry!